plugin.php

```diff
@@ -73,20 +73,23 @@ function register_assets() {
 		$block_info['version']
 	);
 
-	wp_register_script(
-		'wporg-calendar-script',
-		plugin_dir_url( __FILE__ ) . 'build/calendar.js',
-		$frontend_info['dependencies'],
-		$frontend_info['version'],
-		false
-	);
-
-	wp_register_style(
-		'wporg-calendar-style',
-		plugin_dir_url( __FILE__ ) . 'build/calendar.css',
-		array( 'wp-components' ),
-		$frontend_info['version']
-	);
+	// No frontend scripts in the editor
+	if( ! is_admin() ) {
```

Comment:
Technically the editor will still try to enqueue these, even though they're not registered… I don't _see_ any issues, the block still shows up, but this feels like unexpected behavior.

What about conditionally setting up the `$args` in `register_block_type` to not pass through the `script`/`style` args when `is_admin`?

Reply:
Preface: Steve has surface level WordPress knowledge. :)

Mind explaining why they still get enqueued if they are not registered?

Reply:
Because they're added in `register_block_type`, and there's no check [when the block assets are enqueued.](https://github.com/WordPress/WordPress/blob/d6178e1dcf2f663089147793e1858b5a666b3932/wp-includes/script-loader.php#L2498-L2506) Digging deeper, i don't think this actually matters because the code that does [the actual enqueue call](https://github.com/WordPress/WordPress/blob/d6178e1dcf2f663089147793e1858b5a666b3932/wp-includes/class.wp-dependencies.php#L314) checks for registration, so it's a few extra function calls but not actually an issue.

Reply:
Unless it's interfering with something I don't think it's important to bypass loading the front-end script in the editor (is it?). It seems ok to do it, but I think the other change in this PR is the important bit.

Reply:
It's a bit lazy to queue it up. 

Especially since the Gutenberg team is working hard on performance and any console in the log, based on how Gutenberg is architected, is hard to trace.